pegen/parser_generator.py

```diff
@@ -73,10 +73,10 @@ def printblock(self, lines: str) -> None:
             self.print(line)
 
     def collect_todo(self) -> None:
-        done: Set[str] = set()
+        done: List[str] = list()
```

Comment:
Ideally this should remain a `set` to not make the `not in done` part quadratic when considering the loop.

Reply:
But if it's a set and insertion order is not guaranteed, rule generation order is not deterministic, which is our goal here, right? What about both of them being dicts and working with their keys, would that work?

Reply:
You are not iterating over `done` so the insertion order does not matter. Instead of `done = alltodo` you can do `done = set(alltodo)` no?

Reply:
Oh, you mean that only `alltodo` becomes a list and `done` remains a set. That seems reasonable.

Reply:
Yeah, what I was referring is that the `not in done` part is `O(n)` if `done` is a list.